### ISSUE TITLE
Azure Blob Services API - ABSBlobClient.CopyBlob is cumbersome to use

### ISSUE THREAD

**Issue by joandrsn** [microsoft/ALAppExtensions#21894]
`CopyBlob` on `ABSBlobClient` in [Azure Blob Services API](https://github.com/microsoft/ALAppExtensions/tree/main/Modules/System/Azure%20Blob%20Services%20API) has the following defition:
```al
CopyBlob(BlobName: Text; SourceName: Text)
```
Where `BlobName` is the destination and `SourceName` is the source.
A note about the `SourceName` which maps to `x-ms-copy-source` is that when using API version beginning with `2012-02-12`, the Source MUST be the full URL of the object.

So if I want to copy a blob from `test/source.txt` to `test/destination.txt` inside my current `ABSBlobClient`, I have to do the following:
```al
ABSBlobClient := ABSBlobClient.Initialize('myaccount', 'mycontainer', Authorization);
ABSBlobClient.CopyBlob('test/destination.txt', ' https://myaccount.blob.core.windows.net/mycontainer/test/source.txt');
```
The URL I have to build, if I do not want to hardcode variables. The URL can already be generated by the library, but is marked as `Internal`.

As far as I can see, all other functions regarding `ABSBlobClient`, the blobnames are simply blobnames and not full URL's.
This confuses developers using the app as can be seen in #15011, #21671 as the service returns error 400. 

Some suggestions to make it easier to use 
* Create a new function `CopyBlobInContainer` which finds the URL, by using `"ABS URI Helper".ConstructUri` to find the URL of the BLOB
* Highligt that the source MUST be the full URL of the blob

**Comment by pri-kise** [comment]
@joandrsn this looks like a good improvement for the system application, that is totally open source.
And if there are already test for CopyBlobInContainer it should be not this difficult to create new test for a new procedure.
Feel free to create a pull request to fix it on your own.

**Comment by BW-PA** [comment]
I just subscribed to this ticket because I tried using CopyBlob incorrectly too.  After using the other ABS functions which take storage account and container names as parameters I tried to use the CopyBlob function in a similar way.  i.e. just specifying the blob 'name' like this

```
        ABSBlobClient.Initialize('abstemptest', 'blobtest', Authorisation);
        Response := ABSBlobClient.CopyBlob('TestTextFileCopy.txt', 'TestTextFile.txt');
```
which promptly gave me the error 
![image](https://github.com/microsoft/ALAppExtensions/assets/78051738/3bd401e5-7661-4817-80f3-82eb57c2076e)

The error stumped me for a while because ABSBlobClient is initialised to the specific storage container so calling CopyBlob on it in the context of the container is a logical thing to do. It wasn't till I came looking here that I realised I needed to specify a URL for the SourceName parameter.

**Comment by joandrsn** [comment]
I believe this issue was fixed with PR #24481.

The snippet now looks for HTTP in the URL:
https://github.com/microsoft/BCApps/blob/203e4001cb5d1e8e8b51e24fba406921e3a21486/src/System%20Application/App/Azure%20Blob%20Services%20API/src/ABSClientImpl.Codeunit.al#L616-L619

Thank you @ajkauffmann
